quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationIT.java

```diff
@@ -57,10 +66,76 @@ void tokenPropagations() {
+                int statusCode = given()
+                        .contentType("application/json")
+                        .accept("application/json")
+                        .when()
+                        .get("/token_propagation/" + processInstanceId)
+                        .then()
+                        .extract()
+                        .statusCode();
+
+                if (statusCode == 404) {
+                    LOGGER.info("Process instance {} completed successfully (404 - cleaned up)", processInstanceId);
+                    return;
+                }
+
+                Thread.sleep(1000); // Wait 1 second before checking again
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("Interrupted while waiting for process completion", e);
+            } catch (Exception e) {
+                LOGGER.debug("Error checking process state (will retry): {}", e.getMessage());
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ie) {
+                    Thread.currentThread().interrupt();
+                    throw new RuntimeException("Interrupted while waiting for process completion", ie);
+                }
+            }
+        }
+
+        throw new RuntimeException("Process instance " + processInstanceId + " did not complete within " + timeout);
+    }
+
+    private void validateExternalServiceInvocations() {
+        WireMockServer wm = TokenPropagationExternalServicesMock.getInstance();
+
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service1/executeQuery1", "Bearer " + AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service2/executeQuery2", "Bearer " + AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service3/executeQuery3", "Bearer " + SERVICE3_AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service4/executeQuery4", "Bearer " + SERVICE4_AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service5/executeQuery5", "Bearer " + KeycloakServiceMock.KEYCLOAK_ACCESS_TOKEN, 2);
+    }
+
+    private void assertCalledExactlyWithAuth(WireMockServer wm, String url, String expectedAuthHeader, int expectedCalls) {
+        var pattern = WireMock.postRequestedFor(WireMock.urlEqualTo(url));
+        var requests = wm.findAll(pattern);
+        Assertions.assertThat(requests)
+                .as("Expected %s to be called exactly %d times", url, expectedCalls)
+                .hasSize(expectedCalls);
+        for (LoggedRequest req : requests) {
+            Assertions.assertThat(req.getHeader(HttpHeaders.AUTHORIZATION))
+                    .as("Authorization header should match for %s", url)
+                    .isEqualTo(expectedAuthHeader);
```

Comment:
For tracking the obtained result in case of failure, probably we can logged it too:

```suggestion
            Assertions.assertThat(req.getHeader(HttpHeaders.AUTHORIZATION))
                      .as("Expected Authorization header should match for %s but got '%s'", 
                       url, req.getHeader(HttpHeaders.AUTHORIZATION))
                    .isEqualTo(expectedAuthHeader);
```